### PR TITLE
Adds RTC, AON, PRCM driver to CC26x2

### PR DESCRIPTION
**Pull request by refugeesus**
### Pull Request Overview

Add support for RTC, power domain and clock control to the cc26x2 chip.

### Testing Strategy

This PR was tested by initializing dependent peripherals (RFCore from a separate PR) and passing is_enabled() checks. Panic on fail. 

### TODO or Help Wanted

n/a

### Documentation Updated

- [ ] ~~Updated the relevant files in `/docs`, or no updates are required.~~

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

File: chips/cc26x2/src/prcm.rs [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+//!     * Peripheral Power domain
+//!
+//! It also manages the clocks attached to almost every peripheral, which needs to
+//! be enabled before usage.
+//!
+use kernel::common::registers::{ReadOnly, ReadWrite, WriteOnly};
+use kernel::common::StaticRef;
+
+#[repr(C)]
+struct AonWucRegisters {
```
Comment (alevy) [inline chips/cc26x2/src/prcm.rs]: I can't find a reference to this anywhere in the datasheet. Can you add a pointer in the comments?

File: chips/cc26x2/src/rtc.rs [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    }
+
+    pub fn stop(&self) {
+        let regs = &*self.registers;
+        regs.ctl.write(Control::ENABLE::CLEAR);
+
+        regs.sync.get();
+    }
+
+    pub fn sync(&self) {
```
Comment (alevy) [inline chips/cc26x2/src/rtc.rs]: Comment describing what this does (really comments for all of these would be good, but this is the most ambiguous, and not really related to the functionality if the RTC.

File: chips/cc26x2/src/prcm.rs (hunk 1 above)
Comment (refugeesus) [reply chips/cc26x2/src/prcm.rs]: Its `wake-up-control`. These are the same registers as PMCtl registers set in AON. It's needed here for clock access when forcing shutdown. Another not so clear portion of the tech ref manual. If the LF clock is not deactivated (wherever you are sourcing it from), the mcu will not power down correctly. 

I can add comments in to explain this and some reference page numbers

**Comment by alevy** [comment]
@bradjc: relatively little. there is an additional  `sync` method which is used to synchronize between the AON and MCU clocks (why is it in the RTC registers? idk... ask TI...)

Otherwise not much, but I'm personally of a mind to get rid of the cc26xx crate as soon as we have equivalents for the cc26x2 family specifically. It's a crazy enough datasheet that I'm fairly unconfident in any changes we'd make without testing on hardware, and I don't think anyone is actively developing for the CC2650 (sensortag) anymore (nor should they since it doesn't have an MPU)

**Pull request merged**